src/components/Element.js

```diff
@@ -67,7 +71,7 @@ const Element = (type: string, hocOptions: {sourceType?: string} = {}) =>
 
         this._setupEventListeners(element);
 
-        element.mount(this._ref);
+        element.mount(this.props.id ? `#${this.props.id}` : this._ref);
```

Comment:
What's this change needed for?

Seems like `this.props.id` would always point at the same DOM element as `this._ref`.
Is there any case where they could diverge?